src/Symfony/Component/Cache/Adapter/CouchbaseCollectionAdapter.php

```diff
@@ -74,31 +71,32 @@ public static function createConnection(#[\SensitiveParameter] array|string $dsn
                     throw new InvalidArgumentException('Invalid Couchbase DSN: it does not start with "couchbase:".');
                 }
 
-                preg_match($dsnPattern, $server, $matches);
+                $params = parse_url($server);
 
-                $username = $matches['username'] ?: $username;
-                $password = $matches['password'] ?: $password;
-                $protocol = $matches['protocol'] ?: $protocol;
+                $username = $params['user'] ?? $username;
```

Comment:
same for the password:
```suggestion
                $username = isset($params['user']) ? rawurldecode($params['user']) : $username;
```